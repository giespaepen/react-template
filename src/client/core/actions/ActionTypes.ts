export const APP_CONFIGURED: string = "APP_CONFIGURED";
export const APP_INIT: string = "APP_INIT";
export const CLEAR_ERROR: string = "CLEAR_ERROR";
export const CONFIG_RECEIVED: string = "CONFIG_RECEIVED";
export const ERROR_OCCURRED: string = "ERROR_OCCURRED";
export const MESSAGES_HIDE: string = "MESSAGES_HIDE";
export const MESSAGES_HIDE_ALL: string = "MESSAGES_HIDE_ALL";
export const MESSAGES_NEW: string = "MESSAGES_NEW";
export const MESSAGES_PURGE: string = "MESSAGES_PURGE";
export const PROCESS_IDLE: string = "PROCESS_IDLE";
export const PROCESS_STARDED: string = "PROCESS_STARTED";
export const SET_APP_TITLE: string = "SET_APP_TITLE";
export const SET_ERROR: string = "SET_ERROR";
export const SET_PAGE_TITLE: string = "SET_PAGE_TITLE";
