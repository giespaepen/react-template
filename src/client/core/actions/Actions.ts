import { Action } from "redux";

export interface IDefaultAction extends Action {
}

export interface IGenericAction<T> extends IDefaultAction {
    payload: T;
    error?: Error;
}
