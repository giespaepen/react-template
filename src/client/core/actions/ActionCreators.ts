import * as A from "./Actions";
import * as D from "../Domain";
import * as T from "./ActionTypes";
import { Message } from "../components/messages/Properties";

export * from "./creators/AppInitCreators";
export * from "./creators/CreatorUtils";
export * from "./creators/ErrorCreators";
export * from "./creators/MessageCreators";
export * from "./creators/ProcessCreators";
export * from "./creators/TitleCreators";
