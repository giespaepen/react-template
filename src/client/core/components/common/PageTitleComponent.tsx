import * as C from "../../actions/ActionCreators";
import * as React from "react";
import State from "../../state/BaseState";
import { Action } from "redux";
import { Dispatch } from "redux";
import { IPageTitleProperties } from "./Properties";
import { connect } from "react-redux";

/**
 * Statefull component, dispatches a title as page title
 */
@(connect as any)(undefined, PageTitleComponent.mapDispatchToProps)
export default class PageTitleComponent extends React.Component<IPageTitleProperties, State> {

    public static mapDispatchToProps(dispatch: Dispatch<Action>, ownProps: IPageTitleProperties): IPageTitleProperties {
        let props: IPageTitleProperties = Object.assign({}, ownProps);
        props.titleHandler = (title: string) => { dispatch(C.createSetPageTitleAction(title)); };
        return props;
    }

    public render(): any {
        return <span />;
    }

    protected componentDidMount(): void {
        this.props.titleHandler(this.props.title);
    }
}
