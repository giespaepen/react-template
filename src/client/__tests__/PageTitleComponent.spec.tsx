import * as React from "react";
import * as renderer from "react-test-renderer";
import PageTitleComponent from "../core/components/common/PageTitleComponent";
import createTestStore from "../core/util/TestStore";
import { Provider } from "react-redux";
import { SET_PAGE_TITLE } from "../core/actions/ActionTypes";
import { applyMiddleware, compose, createStore, Store } from "redux";

type Action = { type: string, payload: any };

// 
let actionCounter: number = 0;
let lastAction: Action;

test("PageTitleComponent snapshot test", () => {
    // Arrange
    const { reducer, store } = createTestStore({});
    const initialCount: number = reducer.getCounter();
    const expectedAction: string = SET_PAGE_TITLE;

    // Act
    const component = renderer.create(<Provider store={store}><PageTitleComponent title="test" /></Provider>);
    const tree = component.toJSON();
    const actualCount: number = reducer.getCounter();

    // Assert
    expect(tree).toMatchSnapshot();
    expect(actualCount).toBeGreaterThan(initialCount);
    expect(expectedAction).toBe(reducer.getLastAction().type);
});

