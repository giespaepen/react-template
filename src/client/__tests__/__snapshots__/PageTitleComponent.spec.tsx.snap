// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitleComponent snapshot test 1`] = `<span />`;
